airbyte-integrations/connectors/source-mailgun/source_mailgun/source.py

```diff
@@ -0,0 +1,178 @@
+        path: str = "domains"
+        if next_page_token:
+            token: str = next_page_token['url'].rpartition('/')[2]
+            path = f"{path}/{token}"
+
+        return path
+
+
+# Basic incremental stream
+class IncrementalMailgunStream(MailgunStream, ABC):
+
+    @staticmethod
+    def chunk_timestamps_range(start_timestamp: float, interval: Number = None) -> Iterable[Tuple[float]]:
+        """
+        Yield a tuple of beginning and ending timestamps of each day between the start timestamp and now.
+        """
+        interval = interval or 60 * 60 * 24  # 1 day
+        now = time.time()
+        if start_timestamp > now:
+            yield start_timestamp, start_timestamp
+
+        while start_timestamp <= now:
+            end_timestamp = start_timestamp + interval
+            yield start_timestamp, end_timestamp
+            start_timestamp = end_timestamp
+
+
+class Events(IncrementalMailgunStream):
+    # TODO: Event Polling. See https://documentation.mailgun.com/en/latest/api-events.html#event-polling
+
+    cursor_field = "timestamp"
+
+    primary_key = "id"
+
+    def __init__(self, config: Mapping[str, Any], *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Messages are stored for 3 days, so it prevents occasional attempt to read from the start of the Epoch
+        default_shift = 60 * 60 * 24 * 3
+        self.start_timestamp = config.get('timestamp', time.time() - default_shift)
+        self.ascending = True
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> \
+            Mapping[str, Any]:
+        try:
+            current_parsed_timestamp = current_stream_state['timestamp']
+            latest_record_timestamp = latest_record['timestamp']
+            return {'timestamp': max(current_parsed_timestamp, latest_record_timestamp)}
+        except (KeyError, TypeError):
+            return {'timestamp': self.start_timestamp}
+
+    def path(self, *args, next_page_token: Optional[Mapping[str, Any]] = None, **kwargs) -> str:
+        # return super().path(*args, **kwargs) or "events"  # TODO: Requires the CDK update (see MailgunStream.path())
+        path: str = "events"
+        if next_page_token:
+            token: str = next_page_token['url'].rpartition('/')[2]
+            path = f"{path}/{token}"
+
+        return path
+
+    def request_params(
+            self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None,
+            next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(
+            stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token
+        )
+        stream_slice = stream_slice or {
+            "begin": stream_state["timestamp"],
+            "ascending": "yes" if self.ascending else "no",
```

Comment:
is there any reason we make a variable for it? I mean, are we going to have another behavior soon? or just follow KISS?